samples/posix/syslog/src/main.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2024, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#define N_PRIOS 8
+
+int main(void)
+{
+	int prios[N_PRIOS] = {
+		LOG_EMERG,   LOG_ALERT,  LOG_CRIT, LOG_ERR,
+		LOG_WARNING, LOG_NOTICE, LOG_INFO, LOG_DEBUG,
+	};
+
+	openlog("main", LOG_PID | LOG_CONS | LOG_NOWAIT, LOG_LOCAL7);
+	(void)setlogmask(LOG_MASK(-1));
+
+	for (size_t i = 0; i < N_PRIOS; ++i) {
+		syslog(i, "syslog priority %d", prios[i]);
+	}
+
+	closelog();
+
+	/* yield briefly to logging thread */
+	usleep(100000);
+
+	return 0;
+}
```

Comment:
Not necessarily blocking, but important nevertheless :)

I am starting to think it probably doesn't scale all that well to keep adding samples that feel like they are "just" exercising POSIX APIs, and not really showing anything new to the end user. "Of course" syslog should just work, if Zephyr is POSIX compliant, so is there really a point in showing it, especially if no Zephyr-specific tweak is needed?

Surely, having *one* minimal POSIX sample is necessary so that people can quickly get a sense of the Kconfigs they need, or anything else needed to get their code "POSIX-ready", is needed, but not sure we need more of such "minimal" samples (plus, they are costly to run in CI -- and probably paraphrasing existing tests anyway?)

It would probably make more sense to have a much richer sample that effectively uses tons of POSIX APIs to make the point that It Just Works™, once and for all.

Reply:
@kartben - I could potentially move the sample to the existing tests/posix/common suite. As long as it compiles and runs that should be sufficient. Does that work for you?

Reply:
Yup, that would definitely work!